qiskit/circuit/library/n_local/evolved_operator_ansatz.py

```diff
@@ -15,16 +15,262 @@
 from __future__ import annotations
 from collections.abc import Sequence
 
+import typing
+import warnings
+import itertools
 import numpy as np
 
 from qiskit.circuit.library.pauli_evolution import PauliEvolutionGate
 from qiskit.circuit.parameter import Parameter
+from qiskit.circuit.parametervector import ParameterVector
 from qiskit.circuit.quantumregister import QuantumRegister
 from qiskit.circuit.quantumcircuit import QuantumCircuit
 from qiskit.quantum_info import Operator, Pauli, SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from qiskit._accelerate.circuit_library import pauli_evolution
 
 from .n_local import NLocal
 
+if typing.TYPE_CHECKING:
+    from qiskit.synthesis.evolution import EvolutionSynthesis
+
+
+def evolved_operator_ansatz(
+    operators: BaseOperator | Sequence[BaseOperator],
+    reps: int = 1,
+    evolution: EvolutionSynthesis | None = None,
+    insert_barriers: bool = False,
+    name: str = "EvolvedOps",
+    parameter_prefix: str | Sequence[str] = "t",
+    remove_identities: bool = True,
+    flatten: bool | None = None,
+):
+    r"""Construct an ansatz out of operator evolutions.
+
+    For a set of operators :math:`[O_1, ..., O_J]` this circuit is defined as
+
+    .. math::
+
+        \prod_{r=1}^{R} \left( \prod_{j=J}^1 e^{i\theta_{j, r} O_j} \right)
+
+    where the exponentials :math:`exp(i\theta O_j)` are expanded using the product formula
+    specified by ``evolution``.
```

Comment:
A few questions about this. In the formula, ``R`` corresponds to ``reps`` from the docstring? The $\theta_{j, r}$ actually depend on $r$? What is the relation between `evolved_operator_ansatz` and the `PauliEvolutionGate`? And in particular, why in `evolved_operator_ansatz` we have $e^{i\theta O_j}$ and in the definition of `PauliEvolutionGate` we have $e^{-it H}$, are the signs consistent? (My questions really show my ignorance of the subject.)

Reply:
* yes, $R$ is `reps`: I'll add a comment
* `evolved_operator_ansatz` uses layers of `PauliEvolutionGate` to implement a sequence of parameterized evolutions. Indeed they use the _same_ sign, but this is wrongly documented 🙂 
* (you didn't ask but for completeness)`hamiltonian_variational_ansatz` is a special type of `evolved_operator_ansatz` where the operators are set to the commuting groups of a Hamiltonian

Reply:
Added the above in 3e009a5e7